src/commands/profiles/export.ts

```diff
@@ -0,0 +1,286 @@
+      azureConfig.storageName
+    );
+    cli.action.stop();
+
+    // init cosmos client
+    const client = new cosmos.CosmosClient({ endpoint, auth: { key } });
+    const database = client.database(azureConfig.cosmosDatabaseName);
+
+    const selectFromFiscalCode = `SELECT * FROM c WHERE c.fiscalCode = ${fiscalCodeParamName}`;
+    const selectFromMessageId = `SELECT * FROM c WHERE c.messageId = ${messageIdParamName}`;
+    const selectFromRecipientFiscalCode = `SELECT * FROM c WHERE c.recipientFiscalCode = ${fiscalCodeParamName}`;
+
+    cli.action.start("Retrieving user profiles...");
+
+    const profiles = "profiles";
+    const messages = "messages";
+    const notifications = "notifications";
+    const senderServices = "senderServices";
+
+    // define all operations to perform to retrieve all user's data
+    const operations: ReadonlyArray<IOperation> = [
+      // profiles
+      {
+        name: profiles,
+        container: azureConfig.cosmosProfilesContainer,
+        query: selectFromFiscalCode,
+        whereValue: fiscalCode,
+        enableCrossPartitionQuery: false
+      },
+      // messages
+      {
+        name: messages,
+        container: azureConfig.cosmosMessagesContainer,
+        query: selectFromFiscalCode,
+        whereValue: fiscalCode,
+        enableCrossPartitionQuery: false
+      },
+      // notifications
+      {
+        name: notifications,
+        container: azureConfig.cosmosNotificationContainer,
+        query: selectFromFiscalCode,
+        whereValue: fiscalCode,
+        enableCrossPartitionQuery: true
+      },
+      // sender services
+      {
+        name: senderServices,
+        container: azureConfig.cosmosSenderServicesContainer,
+        query: selectFromRecipientFiscalCode,
+        whereValue: fiscalCode,
+        enableCrossPartitionQuery: false
+      }
+    ];
+
+    // execute all operations sequentially
+    const items: ReadonlyArray<IOperationResult> = await sequential(
+      operations,
+      async (op: IOperation) => {
+        cli.action.start(`Retrieving ${op.name}...`);
+        const retrievedItems = await this.getItems(
+          database,
+          op.container,
+          op.query,
+          op.whereValue,
+          op.enableCrossPartitionQuery
+        );
+        cli.action.stop(`${retrievedItems.length} ${op.name} found!`);
+        return { name: op.name, items: retrievedItems };
+      }
+    );
+
+    // dumpData contains all data to be dumped in a zip file
+    // tslint:disable-next-line: readonly-array
+    const dumpData = [...items];
+
+    // get message section
+    const messagesItems = items.find(
+      i => i.name === messages && i.items.length > 0
+    );
+
+    // if there are messages, retrieve all related messages status and messages blob
+    if (messagesItems) {
+      cli.action.start("Retrieving user messages status...");
+      const resultMessagesStatus = await sequential(
```

Comment:
PR lgtm, just note that if you want sequential ops on promise you'd better work using TaskEithers

https://grossbart.github.io/fp-ts-recipes/#/async?a=run-a-list-of-tasks-in-sequence